src_py/cursors.py

```diff
@@ -31,8 +31,90 @@
 The compile() function can convert these string cursors into cursor byte data.
 """
 
-# default pygame black arrow
-arrow = (
+import pygame
+_cursor_id_table = {
+    pygame.SYSTEM_CURSOR_ARROW: "SYSTEM_CURSOR_ARROW",
+    pygame.SYSTEM_CURSOR_IBEAM: "SYSTEM_CURSOR_IBEAM",
+    pygame.SYSTEM_CURSOR_WAIT: "SYSTEM_CURSOR_WAIT",
+    pygame.SYSTEM_CURSOR_CROSSHAIR: "SYSTEM_CURSOR_CROSSHAIR",
+    pygame.SYSTEM_CURSOR_WAITARROW: "SYSTEM_CURSOR_WAITARROW",
+    pygame.SYSTEM_CURSOR_SIZENWSE: "SYSTEM_CURSOR_SIZENWSE",
+    pygame.SYSTEM_CURSOR_SIZENESW: "SYSTEM_CURSOR_SIZENESW",
+    pygame.SYSTEM_CURSOR_SIZEWE: "SYSTEM_CURSOR_SIZEWE",
+    pygame.SYSTEM_CURSOR_SIZENS: "SYSTEM_CURSOR_SIZENS",
+    pygame.SYSTEM_CURSOR_SIZEALL: "SYSTEM_CURSOR_SIZEALL",
+    pygame.SYSTEM_CURSOR_NO: "SYSTEM_CURSOR_NO",
+    pygame.SYSTEM_CURSOR_HAND: "SYSTEM_CURSOR_HAND",
+}
+
+class Cursor(object):
+    def __init__(self, *args):
+        if len(args) == 0:
+            self.type = "system"
+            self.data = (pygame.SYSTEM_CURSOR_ARROW,)
+        elif len(args) == 1 and args[0] in list(_cursor_id_table.keys()):
+            self.type = "system"
+            self.data = (args[0],)
+        elif len(args) == 1 and isinstance(args[0], Cursor):
+            self.type = args[0].type
+            self.data = args[0].data
+        elif len(args) == 2 and len(args[0]) == 2 and isinstance(args[1], pygame.Surface):
+            self.type = "color"
+            self.data = tuple(args)
+        elif len(args) == 4 and len(args[0]) == 2 and len(args[1]) == 2:
+            self.type = "bitmap"
+            self.data = tuple([tuple(arg) for arg in args])
+        else:
+            raise TypeError("Arguments must match a cursor specification")
+            
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, index):
+        return self.data[index]
+
+    def __eq__(self, other):
+        return isinstance(other, Cursor) and self.data == other.data
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash(tuple([self.type] + [data for data in self.data]))
+
+    def __repr__(self):
+        if self.type == "system":
+            id_string = _cursor_id_table.get(self.data[0], "constant lookup error")
+            return "Cursor<system_cursor(" + id_string +")>"
```

Comment:
I know the rest of pygame doesn't do this well... but repr(obj) is supposed to try and make an object that is constructable. ie. `eval(repr(obj)) == obj`, failing that it has the "<>" surrounding it to say "this is not how you construct it but sort of is what it's about".

I'm wondering if keyword arguments would make this easier.

Reply:
I was trying to imitate other pygame objects like Rect or Surface but I put the "<>" in the wrong place. At the least I'll fix that, move it to this: `<Cursor(type: system, constant: SYSTEM_CURSOR_ARROW)>`

I don't know if a more broad `eval(repr(obj)) == obj` is practical for cursor objects, because they could contain surfaces, and while bringing all the data of a surface into a string and back is possible, it's not very pretty to look at.

Reply:
> <Cursor(type: system, constant: SYSTEM_CURSOR_ARROW)>
Looks good to me.

Reply:
I've made the changes locally, I just don't want to take up build minutes by pushing a commit.
I don't know how to skip ci on a commit.

Reply:
You can add the text in your commit:
```
[skip ci]
```